src/scenarios.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+import csv
+
+import numpy as np
+from itertools import product
+
+from sim import simulate as sim
+
+#TODO add charger_density once supported in sim.py
+def genScenarios(battery_cap_kwh,nondepot_charger_rate,): #charger_density):
+    '''Generates a list of tuples of scenarios
+        args: all tuples of a start (inclusive), end (exclusive), and step 
+        size values. i.e. battery_cap_kwh=(125,700,100) 
+    '''
+    #TODO include error handling
+    bat_range = np.arange(*battery_cap_kwh)
+    non_depot_range = np.arange(*nondepot_charger_rate)
+    #charger_density = np.arange(*charger_density)
+
+    scen_opts = [bat_range,non_depot_range,]# charger_density]
+    scens = list(product(*scen_opts))
+    return scens
+
+def main():
+  parser = argparse.ArgumentParser(description='TODO')
+  parser.add_argument('parsed_gtfs_prefix', type=str, help='TODO')
+  parser.add_argument('osm_data',           type=str, help='TODO')
+  parser.add_argument('depots_filename',    type=str, help='TODO')
+  parser.add_argument('output_filename',    type=str, help='TODO')
+  parser.add_argument('-b','--battery-cap-kwh',      nargs=3,type=int,help='TODO')
+  parser.add_argument('-c','--nondepot-charger-rate',nargs=3,type=int,help='TODO')
+  #TODO add when supported by sim.py
+  #parser.add_argument('-d','--charger-density',  nargs=3,type=int,help='TODO')
+  parser.add_argument('-s','--save_csv',
+    nargs='?',const='scenarios',type=str,help='TODO')
+  args = parser.parse_args()
+
+  #extract scenario-related arguments and generate scenario array
+  list_args = set(['battery_cap_kwh','nondepot_charger_rate',])#'charger_density'])
+  scen_args = {k:v for (k,v) in vars(args).items() if k in list_args}
+  scenarios = genScenarios(**scen_args)
+
+  if args.save_csv is not None:
+    #TODO sync up with output_filename
```

Comment:
Fix TODO.

Reply:
It is unclear to me how you set the `const=` value of `parser.add_argument('-s','--save_csv',const= <VALUE OF output_filename> . . )` before `parser.parse_args()` is called. The [documenation](https://docs.python.org/3.7/library/argparse.html#argparse.ArgumentParser.add_argument) has not been helpful.